pkg/traces/config.go

```diff
@@ -141,6 +144,35 @@ type InstanceConfig struct {
 // YAML to the text "<secret>".
 type ReceiverMap map[string]interface{}
 
+// UnmarshalYAML implements yaml.Unmarshaler.
+func (r *ReceiverMap) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain ReceiverMap
+	if err := unmarshal((*plain)(r)); err != nil {
+		return err
+	}
+
+	protocols := []string{protocolHTTP, protocolGRPC}
+	// enable include_metadata by default if receiver is OTLP
+	for k := range *r {
+		if strings.HasPrefix(k, otlpReceiverName) {
+			// for http and grpc receivers, include_metadata is set to true by default
+			for _, p := range protocols {
+				if cfg, ok := (*r)[k].(map[interface{}]interface{})["protocols"].(map[interface{}]interface{})[p]; ok {
```

Comment:
this is quite the line. let's break it up for clarity. this code panics if `(*r)[k]` is not a `(map[interface{}]interface{})`?

let's do these casts once and store them in variables to be used later. will help with code clarity.

Reply:
Simplified casting. Done!